src/main/java/org/apache/commons/math4/util/FastMath.java

```diff
@@ -871,7 +879,11 @@ public static double exp(double x) {
      * @param hiPrec extra bits of precision on output (To Be Confirmed)
      * @return exp(x)
      */
-    private static double exp(double x, double extra, double[] hiPrec) {
+    @SuppressWarnings({"index:array.access.unsafe.low", "index:array.access.unsafe.high"}) /*
+    #1: EXP_INT_TABLE_MAX_INDEX is @NonNegative and @LTLengthOf(value = {"ExpIntTable.EXP_INT_TABLE_A","ExpIntTable.EXP_INT_TABLE_B"}, offset = {"749", "749"}) as annotated
```

Comment:
This doesn't justify why the access is safe - how do you know that `intVal` is a valid offset? Isn't it just an unannotated argument, which could therefore take on any value? That doesn't seem safe.

Reply:
Instead, I think you need to make it more clear that the `if` and `return` statements that come first guarantee that `intVal` is in the range `[-709, 709]`, which, given that the length of `EXP_INT_TABLE_A` and `EXP_INT_TABLE_B` are `@MinLen(1449)` (which I think they are?) should be safe.

Reply:
(1449 = 750 + 709, by the way. 750 is the value of `EXP_INT_TABLE_MAX_INDEX`, I believe).